go.mod

```diff
@@ -1,76 +1,68 @@
 	github.com/xeipuuv/gojsonschema v1.2.0 // indirect
 	go.uber.org/zap v1.11.0 // indirect
-	k8s.io/api v0.0.0
-	k8s.io/apimachinery v0.0.0
-	k8s.io/apiserver v0.0.0
-	k8s.io/client-go v0.0.0
-	k8s.io/component-base v0.0.0
+	k8s.io/api v0.18.0
+	k8s.io/apimachinery v0.18.0
+	k8s.io/apiserver v0.18.0
+	k8s.io/client-go v0.18.0
+	k8s.io/component-base v0.18.0
 	k8s.io/klog v1.0.0
-	k8s.io/kubectl v0.0.0
-	k8s.io/kubernetes v0.0.0
-	k8s.io/utils v0.0.0-20191114184206-e782cd3c129f
+	k8s.io/kubectl v0.18.0
+	k8s.io/kubernetes v1.18.0
+	k8s.io/utils v0.0.0-20200324210504-a9aa75ae1b89
 )
 
 replace (
-	bitbucket.org/ww/goautoneg => github.com/munnerz/goautoneg v0.0.0-20120707110453-a547fc61f48d
-	github.com/golang/glog => github.com/openshift/golang-glog v0.0.0-20190322123450-3c92600d7533
-	github.com/onsi/ginkgo => github.com/openshift/onsi-ginkgo v1.2.1-0.20190125161613-53ca7dc85f60
-	k8s.io/api => k8s.io/api v0.0.0-20191122220107-b5267f2975e0
-	k8s.io/apiextensions-apiserver => k8s.io/apiextensions-apiserver v0.0.0-20191122222427-64482ea217ff
-	k8s.io/apimachinery => k8s.io/apimachinery v0.0.0-20191121175448-79c2a76c473a
-	k8s.io/apiserver => k8s.io/apiserver v0.0.0-20191122221311-9d521947b1e1
-	k8s.io/cli-runtime => k8s.io/cli-runtime v0.0.0-20191122222818-9150eb3ded31
-	k8s.io/client-go => k8s.io/client-go v0.0.0-20191122220542-ed16ecbdf3a0
-	k8s.io/cloud-provider => k8s.io/cloud-provider v0.0.0-20191122224014-59d76ad41876
-	k8s.io/cluster-bootstrap => k8s.io/cluster-bootstrap v0.0.0-20191122223827-289de4a64c1c
-	k8s.io/code-generator => k8s.io/code-generator v0.0.0-20191121175249-e95606b614f0
-	k8s.io/component-base => k8s.io/component-base v0.0.0-20191122220729-2684fb322cb9
-	k8s.io/cri-api => k8s.io/cri-api v0.0.0-20191121183020-775aa3c1cf73
-	k8s.io/csi-translation-lib => k8s.io/csi-translation-lib v0.0.0-20191122224157-518651485193
-	k8s.io/kube-aggregator => k8s.io/kube-aggregator v0.0.0-20191122221605-1e8d331e4dcc
-	k8s.io/kube-controller-manager => k8s.io/kube-controller-manager v0.0.0-20191122223648-5cfd5067047c
-	k8s.io/kube-proxy => k8s.io/kube-proxy v0.0.0-20191122223145-16f2c0c680a0
-	k8s.io/kube-scheduler => k8s.io/kube-scheduler v0.0.0-20191122223507-ac656df73673
-	k8s.io/kubectl => k8s.io/kubectl v0.0.0-20191122225023-1e3c8b70f494
-	k8s.io/kubelet => k8s.io/kubelet v0.0.0-20191122223325-9316382755ad
-	k8s.io/kubernetes => k8s.io/kubernetes v1.17.0-rc.1
-	k8s.io/legacy-cloud-providers => k8s.io/legacy-cloud-providers v0.0.0-20191122224431-860df69ff5cc
-	k8s.io/metrics => k8s.io/metrics v0.0.0-20191122222628-19ed227de2b6
-	k8s.io/sample-apiserver => k8s.io/sample-apiserver v0.0.0-20191122221846-294c70c3d5d4
+	// for whatever weird reason go mod would not let me set openshift/api to this level in the require section
+	github.com/openshift/api => github.com/openshift/api v0.0.0-20200326160804-ecb9283fe820
+    // these are needed since k8s.io/kubernetes cites v0.0.0 for each of these k8s deps in its go.mod
```

Comment:
could this be more limited just to what's being used in required section?

Reply:
Nope.  I had to add these one by one, otherwise I got errors like

```
go: k8s.io/kubernetes@v1.18.0 requires
	k8s.io/apiextensions-apiserver@v0.0.0: reading k8s.io/apiextensions-apiserver/go.mod at revision v0.0.0: unknown revision v0.0.0
```

Which is what I was referring to in the comment above.

Even when I tried moving each of these to the require section, and remove them from the replace section, I got the same errors, like 

```
go: k8s.io/kubernetes@v1.18.0 requires
	k8s.io/api@v0.0.0: reading k8s.io/api/go.mod at revision v0.0.0: unknown revision v0.0.0
```

Reply:
ok, thanks for the detail